### PR TITLE
clang-tidy workflow

### PR DESCRIPTION
**Pull request by andiwand**

### CODE REVIEW

File: .github/workflows/tidy.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    steps:
+      - run: sudo apt install clang-tidy
+
+      - name: checkout
+        uses: actions/checkout@v2
+
+      - run: clang-tidy --version
+
+      - name: cmake
```
Comment (TomTasche) [inline .github/workflows/tidy.yml:17]: why is that necessary to check format?

Comment (andiwand) [reply .github/workflows/tidy.yml:17]: cmake generates a file that contains all the build commands. clang-tidy is using it to get the flags and includes.
